src/meta/src/manager/catalog/mod.rs

```diff
@@ -797,13 +797,15 @@ impl CatalogManager {
         Ok(())
     }
 
-    fn assert_table_creating(tables: &BTreeMap<TableId, Table>, table: &Table) {
-        if let Some(t) = tables.get(&table.id)
-            && let Ok(StreamJobStatus::Creating) = t.get_stream_job_status()
-        {
+    fn check_table_creating(tables: &BTreeMap<TableId, Table>, table: &Table) -> MetaResult<()> {
```

Comment:
Cc @kwannoel , Before the table in `Foreground` is marked as created, if recovery occurs, it will be cleaned up by `clean_dirty_tables`. Changed this function to check and return an error if it's cleaned. 
The corner case was found in this run: https://buildkite.com/risingwavelabs/pull-request/builds/43296#018df426-a3da-48cc-af24-f09eb061a320